### ISSUE TITLE
Remaining words issue with repeated letters

### ISSUE THREAD

**Issue by Cirzen** [jakearchibald/wordle-analyzer#13]
Spoilers for 20th June 
Wordle 4/6

🟨⬜⬜⬜⬜
⬜⬜🟨⬜🟩
⬜🟨⬜⬜🟩
🟩🟩🟩🟩🟩

https://wordle-analyzer.com/?seed=44&guesses=uhqcfrkewqnudnlbgyrr&hm=1
In the above analysis, after turn 3, the AI suggests that "UNFIT" is a valid possibility, even though the previous word score has ruled out an "I" in the 4th position

**Comment by jakearchibald** [comment]
> Spoilers for 20th June

Thank you! I've managed to spoil games for myself by debugging issues here. However, I've already played today's, so let's go…

> In the above analysis, after turn 3, the AI suggests that "UNFIT" is a valid possibility, even though the previous word score has ruled out an "I" in the 4th position

Ohh, thanks for spotting this. I'll fix it up.

For my own documentation:

![Screen Shot 2022-06-20 at 13 23 52](https://user-images.githubusercontent.com/93594/174601715-76c5bafb-75aa-439a-ad40-0cd1543f7b1f.png)

Right now, the algorithm comes up with the following rules:

- L is not in the word
- I is in the word
- I is not in the 2nd position
- M is not in the word
- There is only 1 I in the word
- The last letter is T

But if the 4th letter was I, then it'd be a green square (and the 2nd would be blank), so there's an additional rule, "I is not in the 4th position".
